python/dgl/cuda/nccl.py

```diff
@@ -1,230 +1,184 @@
-"""API creating NCCL communicators."""
+"""API wrapping NCCL primitives."""
 
-from .. import backend as F
-from .._ffi.function import _init_api
+import torch
+import torch.distributed as dist
 
-_COMM_MODES_MAP = {"remainder": 0}
 
+def sparse_all_to_all_push(idx, value, partition):
+    """Perform an all-to-all-v operation, where by all processors send out
+    a set of indices and corresponding values. Indices and values,
+    corresponding to the current process, will copied into the output
+    arrays.
 
-class UniqueId(object):
-    """Class for allowing python code to create and communicate NCCL Unique
-    IDs, needed for creating communicators.
-    """
+    Note: This method requires 'torch.distributed.is_initialized()' and
```

Comment:
What would happen if the backend is not `nccl`? Shall we raise an error here?

Reply:
Added an assertation.